File: encryptors/noencryption/noencryption.go
package noencryption

// No encryption source
//
// This is the simplest source type
type NoEncryptionSource struct {
}

func (p NoEncryptionSource) ID() string {
	return "noencryption$$$$"
}

// v2 specific
func (p NoEncryptionSource) Encrypt(b []byte) ([]byte, error) {
	return b, nil
}

// v2 specific
func (p NoEncryptionSource) Decrypt(b []byte) ([]byte, error) {
	return b, nil
}

File: rust/src/encryptors/noencryption.rs
use crate::autoencrypted::AutoEncryptor;

pub struct NoEncryptionSource;

impl AutoEncryptor for NoEncryptionSource {
    fn id(&self) -> String {
        "noencryption$$$$".to_string()
    }

    fn encrypt(&mut self, data: &[u8]) -> Result<Vec<u8>, crate::Error> {
        Ok(data.to_vec())
    }

    fn decrypt(&mut self, data: &[u8]) -> Result<Vec<u8>, crate::Error> {
        Ok(data.to_vec())
    }
}
